src/sctools/platform.py

```diff
@@ -0,0 +1,71 @@
+from . import fastq, bam
+import argparse
+
+
+class PlatformBase:
+    """Abstract base class for platform specific metadata"""
+
+    @classmethod
+    def get_tag(cls, sequencing_read):
+        raise NotImplementedError
+
+    @classmethod
+    def tag_bamfile(cls, input_bamfile_name, output_bamfile_name, files_with_tags):
+        """Extracts tags from fastq files_with_tags, attaches them to input_bamfile_name and writes
+        the result to output_bamfile_name
+
+        :param str input_bamfile_name: input bam
+        :param str output_bamfile_name: output bam
+        :param dict files_with_tags: dict mapping {filenames: tags}
+        """
+        tag_generators = []
+        for k, v in files_with_tags.items():
+            tag_generators.append(fastq.TagGenerator(cls.get_tag(k), files=v))
```

Comment:
The docstring says the key is the filename and the value is the tag, opposite of what is happening here.